lightning/src/ln/channelmanager.rs

```diff
@@ -7792,22 +7817,34 @@ where
 		inbound_payment::get_payment_preimage(payment_hash, payment_secret, &self.inbound_payment_key)
 	}
 
-	/// Creates a one-hop blinded path with [`ChannelManager::get_our_node_id`] as the introduction
-	/// node.
-	fn create_one_hop_blinded_path(&self) -> BlindedPath {
+	/// Creates a blinded path by delegating to [`MessageRouter::create_blinded_paths`].
+	///
+	/// Errors if the `MessageRouter` errors or returns an empty `Vec`.
+	fn create_blinded_path(&self) -> Result<BlindedPath, ()> {
+		let recipient = self.get_our_node_id();
 		let entropy_source = self.entropy_source.deref();
 		let secp_ctx = &self.secp_ctx;
-		BlindedPath::one_hop_for_message(self.get_our_node_id(), entropy_source, secp_ctx).unwrap()
+
+		let peers = self.per_peer_state.read().unwrap()
+			.iter()
+			.filter(|(_, peer)| peer.lock().unwrap().latest_features.supports_onion_messages())
+			.map(|(node_id, _)| *node_id)
+			.collect::<Vec<_>>();
+
+		self.router
+			.create_blinded_paths(recipient, peers, entropy_source, secp_ctx)
+			.and_then(|paths| paths.into_iter().next().ok_or(()))
 	}
 
-	/// Creates a one-hop blinded path with [`ChannelManager::get_our_node_id`] as the introduction
-	/// node.
-	fn create_one_hop_blinded_payment_path(
-		&self, payment_secret: PaymentSecret
-	) -> (BlindedPayInfo, BlindedPath) {
+	/// Creates multi-hop blinded payment paths for the given `amount_msats` by delegating to
+	/// [`Router::create_blinded_payment_paths`].
+	fn create_blinded_payment_paths(
+		&self, amount_msats: u64, payment_secret: PaymentSecret
+	) -> Result<Vec<(BlindedPayInfo, BlindedPath)>, ()> {
 		let entropy_source = self.entropy_source.deref();
 		let secp_ctx = &self.secp_ctx;
 
+		let first_hops = self.list_usable_channels();
 		let payee_node_id = self.get_our_node_id();
 		let max_cltv_expiry = self.best_block.read().unwrap().height() + LATENCY_GRACE_PERIOD_BLOCKS;
```

Comment:
Looks like this doesn't allow for the router adding the shadow offset to the CLTV during route construction. Will probably want to add `MAX_SHADOW_CLTV_EXPIRY_DELTA_OFFSET` (from router.rs) to this value.

Reply:
Hmm... but the `BlindedPath` is created by the recipient, and the `Route` is created by the payer. Is that value common across implementations? Or just something we expect to be good enough?

Reply:
True. I think it will be good enough, but this does mean that our router should be taking `max_cltv_expiry` into account and currently isn't :/

Reply:
> Looks like this doesn't allow for the router adding the shadow offset to the CLTV during route construction. Will probably want to add `MAX_SHADOW_CLTV_EXPIRY_DELTA_OFFSET` (from router.rs) to this value.

Thinking more, currently the way we set `max_cltv_expiry`here  doesn't allow for ~any CLTV delta along the route, not just the shadow offset. So I think the value we set here should increase even further, actually.

Reply:
> True. I think it will be good enough, but this does mean that our router should be taking `max_cltv_expiry` into account and currently isn't :/

Ok took me a minute to reload all the context here. `max_cltv_expiry` isn't part of `BlindedPayInfo` so there's no way to take it into account when pathfinding. This confused me originally so I opened https://github.com/lightning/bolts/pull/1069/ to clarify it. But I don't think we should stop using the shadow offset in pathfinding just because invoice expiries tend to be short, so I'm not exactly sure what the resolution is here.

Reply:
Discussed offline to add `CLTV_FAR_FAR_AWAY`.